### PR TITLE
backup: Add rawkv ttl expired data metric

### LINKED ISSUE #13436: Add TTL expired count metric during backup
### Is your feature request related to a problem? Please describe:

1. TTL can be specified in tikv raw mode during in [`raw_put`](https://github.com/tikv/tikv/blob/4f717716a515a8047d91f4cbeef219f41c20dc2c/src/storage/mod.rs#L1759-L1767) or [`raw_batch_put`](https://github.com/tikv/tikv/blob/4f717716a515a8047d91f4cbeef219f41c20dc2c/src/storage/mod.rs#L1835-L1842)
2. Duraing raw backup, the ttl expired data will not be saved in backup files, which may bring confusion to users, as the user is seeing different `total-kvs` between two backup operation even if no data is newly inserted.

### Describe the feature you'd like:

Add a counter statistics of ttl expired data during rawkv backup to indicate the the expired data count.

### PR DESCRIPTION
**Pull request by haojinming**
### What is changed and how it works?

Issue Number: Close #13436 

What's Changed:

```
Add rawkv ttl expired data metric
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
ttl expired metrics

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by zeminzhou — APPROVED** [review]
LGTM～

File: components/backup/src/utils.rs [hunk 1]
```diff
@@ -556,4 +566,92 @@ pub mod tests {
...
+                b"a".to_vec(),
+                100,
+                false,
+                true,
+                false,
+            ),
+        ];
+
+        for (idx, (api_ver, key, value, expire_ts, is_delete, expect_valid, expect_ttl_expire)) in
+            test_cases.iter().enumerate()
```
Comment (pingyu) [inline components/backup/src/utils.rs:641]: Nit: Use `test_cases.into_iter().enumerate()`. Then we can use `expire_ts` other than `*expire_ts`.

File: metrics/grafana/tikv_details.json [hunk 2]
```diff
@@ -40363,6 +40363,109 @@
...
+          "points": false,
+          "renderer": "flot",
+          "seriesOverrides": [],
+          "spaceLength": 10,
+          "stack": false,
+          "steppedLine": false,
+          "targets": [
+            {
+              "exemplar": true,
+              "expr": "tikv_backup_raw_expired_count",
```
Comment (pingyu) [inline metrics/grafana/tikv_details.json:40414]: Should use `rate` to get the metric per second.
Also should support selectors `k8s_cluster`, `tidb_cluster`, and `instance`.

File: components/backup/src/utils.rs (hunk 1 above)
Comment (haojinming) [reply components/backup/src/utils.rs]: Done， PTAL~

File: components/backup/src/endpoint.rs [hunk 3]
```diff
@@ -1730,14 +1749,25 @@ pub mod tests {
             assert!(resp.has_error());
             return false;
         }
+        let current_expire_cnt = BACKUP_RAW_EXPIRED_COUNT.get();
```
Comment (pingyu) [inline components/backup/src/endpoint.rs:1752]: Nit: Add blank lines to separate logical block for more readability.

**Review by pingyu — APPROVED** [review]
LGTM~

**Review by 3pointer — APPROVED** [review]

**Pull request merged**